setup/flex.rst

```diff
@@ -0,0 +1,173 @@
+
+.. code-block:: terminal
+
+    $ cd my-project/
+    $ composer require mailer
+
+If you execute that command in a Symfony application which doesn't use Flex,
+you'll see a Composer error explaining that ``mailer`` is not a valid package
+name. However, if the application has Symfony Flex installed, that command ends
+up installing and enabling the SwiftmailerBundle, which is the best way to
+integrate Swiftmailer, the official mailer for Symfony applications.
+
+When Symfony Flex is installed in the application and you execute ``composer require``,
+the application makes a request to Symfony Flex servers before trying to install
+the package with Composer:
+
+* If there's no information about that package, Flex server returns nothing and
+  the package installation follows the usual procedure based on Composer;
+* If there's special information about that package, Flex returns it in a file
+  called "recipe" and the application uses it to decide which package to install
+  and which automated tasks to run after the installation.
+
+In the above example, Symfony Flex asks about the ``mailer`` package and the
+Symfony Flex server detects that ``mailer`` is in fact an alias for SwiftmailerBundle
+and returns the "recipe" for it.
+
+Symfony Flex Recipes
+~~~~~~~~~~~~~~~~~~~~
+
+Recipes are defined in a ``manifest.json`` file and can contain any number of
+other files and directories. For example, this is the ``manifest.json`` for SwiftmailerBundle:
+
+.. code-block:: javascript
+
+    {
+        "bundles": {
+            "Symfony\\Bundle\\SwiftmailerBundle\\SwiftmailerBundle": ["all"]
+        },
+        "copy-from-recipe": {
+            "config/": "%CONFIG_DIR%/"
+        },
+        "env": {
+            "MAILER_URL": "smtp://localhost:25?encryption=&auth_mode="
+        },
+        "aliases": ["mailer", "mail"]
+    }
+
+The ``alias`` option allows Flex to install packages using short and easy to
+remember names (``composer require mailer`` vs ``composer require symfony/swiftmailer-bundle``).
+The ``bundles`` option tells Flex in which environments should this bundle be
+enabled automatically (``all`` in this case). The ``env`` option makes Flex to
+add new environment variables to the application. Finally, the ``copy-from-recipe``
+option allows the recipe to copy files and directories into your application.
+
+The instructions defined in this ``manifest.json`` file are also used by Symfony
+Flex when uninstalling dependencies (e.g. ``composer remove mailer``) to undo
+all changes. This means that Flex can remove the SwiftmailerBundle from the
+application, delete the ``MAILER_URL`` environment variable and any other file
+and directory created by this recipe.
+
+Symfony Flex recipes are contributed by the community and they are stored in
+two public repositories:
+
+* `Main recipe repository`_, is a curated list of recipes for high quality and
+  maintained packages. Symfony Flex only looks for in this repository by default.
```

Comment:
Are we sure? I thought *"look for"* was the right verb here ... so "looks for in ..." instead of "looks in ..."  In any case, I've made the change you proposed.